src/main/java/org/medicmobile/webapp/mobile/MrdtSupport.java

```diff
@@ -37,7 +37,8 @@ String process(int requestCode, int resultCode, Intent i) {
 				try {
 					byte[] data = i.getByteArrayExtra("data");
 					String base64data = Base64.encodeToString(data, Base64.NO_WRAP);
-					return safeFormat("angular.element(document.body).injector().get('AndroidApi').v1.mrdtResponse('\"%s\"')", base64data);
+					long timeTaken = i.getLongExtra("timeTaken", 0);
+					return safeFormat("const api = angular.element(document.body).injector().get('AndroidApi'); try { api.v1.mrdtTimeTakenResponse('\"%s\"'); } catch(ex) {} api.v1.mrdtResponse('\"%s\"')", String.valueOf(timeTaken), base64data);
```

Comment:
Can you just check if `api.v1.mrdtTimeTakenResponse` is defined? Instead of doing the try/catch?